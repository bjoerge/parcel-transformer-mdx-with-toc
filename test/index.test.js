const path = require('path')
const index = require('../.')
const MDXAsset = require('../MDXAsset')

let value

describe('index', () => {
  it('should add asset types', () => {
    const bundler = {addAssetType: jest.fn()}
    const {calls} = bundler.addAssetType.mock
    const mdxAssetPath = require.resolve('../MDXAsset')

    index(bundler)

    expect(calls.length).toEqual(2)
    expect(calls[0][0]).toBe('md')
    expect(calls[0][1]).toBe(mdxAssetPath)
    expect(calls[1][0]).toEqual('mdx')
    expect(calls[1][1]).toEqual(mdxAssetPath)
  })
})

describe('MDXAsset', () => {
  beforeEach(async () => {
    const asset = new MDXAsset(path.resolve(__dirname, './Markdown.mdx'), {
      rootDir: __dirname,
    })

    const processed = await asset.process()
    const jsx = processed.find(({type}) => type === 'js')
    value = jsx.value
  })

  it('should render markdown with MDX.', () => {
    expect(value).toMatchSnapshot()
  })

  it('should render components with MDX.', () => {
    expect(value).toMatchSnapshot()
  })
})
