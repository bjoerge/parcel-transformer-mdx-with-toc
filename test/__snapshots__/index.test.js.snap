// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MDXAsset should render components with MDX. 1`] = `
"
        import React from 'react';
        import { mdx } from '@mdx-js/react'

        export default (function() {/* @jsx mdx */
import Component from './Component'

const makeShortcode = name => function MDXDefaultShortcode(props) {
  console.warn(\\"Component \\" + name + \\" was not imported, exported, or provided by MDXProvider as global scope\\")
  return <div {...props}/>
};

const layoutProps = {
  
};
const MDXLayout = \\"wrapper\\"
function MDXContent({
  components,
  ...props
}) {
  return <MDXLayout {...layoutProps} {...props} components={components} mdxType=\\"MDXLayout\\">

    <h1 {...{
      \\"id\\": \\"test\\"
    }}>{\`Test\`}</h1>
    <Component mdxType=\\"Component\\">component</Component>
    </MDXLayout>;
}

;
MDXContent.isMDXComponent = true;
return MDXContent})();

        export const Toc = (function() {/* @jsx mdx */


const makeShortcode = name => function MDXDefaultShortcode(props) {
  console.warn(\\"Component \\" + name + \\" was not imported, exported, or provided by MDXProvider as global scope\\")
  return <div {...props}/>
};

const layoutProps = {
  
};
const MDXLayout = \\"wrapper\\"
function MDXContent({
  components,
  ...props
}) {
  return <MDXLayout {...layoutProps} {...props} components={components} mdxType=\\"MDXLayout\\">
    <ul>
      <li parentName=\\"ul\\"><a parentName=\\"li\\" {...{
          \\"href\\": \\"#test\\"
        }}>{\`Test\`}</a></li>
    </ul>
    </MDXLayout>;
}

;
MDXContent.isMDXComponent = true;
return MDXContent})()
        "
`;

exports[`MDXAsset should render markdown with MDX. 1`] = `
"
        import React from 'react';
        import { mdx } from '@mdx-js/react'

        export default (function() {/* @jsx mdx */
import Component from './Component'

const makeShortcode = name => function MDXDefaultShortcode(props) {
  console.warn(\\"Component \\" + name + \\" was not imported, exported, or provided by MDXProvider as global scope\\")
  return <div {...props}/>
};

const layoutProps = {
  
};
const MDXLayout = \\"wrapper\\"
function MDXContent({
  components,
  ...props
}) {
  return <MDXLayout {...layoutProps} {...props} components={components} mdxType=\\"MDXLayout\\">

    <h1 {...{
      \\"id\\": \\"test\\"
    }}>{\`Test\`}</h1>
    <Component mdxType=\\"Component\\">component</Component>
    </MDXLayout>;
}

;
MDXContent.isMDXComponent = true;
return MDXContent})();

        export const Toc = (function() {/* @jsx mdx */


const makeShortcode = name => function MDXDefaultShortcode(props) {
  console.warn(\\"Component \\" + name + \\" was not imported, exported, or provided by MDXProvider as global scope\\")
  return <div {...props}/>
};

const layoutProps = {
  
};
const MDXLayout = \\"wrapper\\"
function MDXContent({
  components,
  ...props
}) {
  return <MDXLayout {...layoutProps} {...props} components={components} mdxType=\\"MDXLayout\\">
    <ul>
      <li parentName=\\"ul\\"><a parentName=\\"li\\" {...{
          \\"href\\": \\"#test\\"
        }}>{\`Test\`}</a></li>
    </ul>
    </MDXLayout>;
}

;
MDXContent.isMDXComponent = true;
return MDXContent})()
        "
`;
